module/behaviour/planning/SimpleWalkPathPlanner/src/SimpleWalkPathPlanner.cpp

```diff
@@ -144,97 +149,131 @@ namespace module::behaviour::planning {
+                        emit(std::make_unique<StopCommand>(subsumptionId));
+                        return;
+
+                    case message::behaviour::MotionCommand::Type::DIRECT_COMMAND: walkDirectly(); return;
+
+                    case message::behaviour::MotionCommand::Type::BALL_APPROACH:
+                        determineSimpleWalkPath(ball, field, sensors, kickPlan, fieldDescription);
+                        return;
 
-                Eigen::Affine3d Htw(sensors.Htw);
-
-                auto now = NUClear::clock::now();
-                float timeSinceBallSeen =
-                    std::chrono::duration_cast<std::chrono::nanoseconds>(now - timeBallLastSeen).count()
-                    * (1.0f / std::nano::den);
-
-
-                Eigen::Vector3d rBWw_temp(ball.position.x(), ball.position.y(), fieldDescription.ball_radius);
-                rBWw     = timeSinceBallSeen < search_timeout ? rBWw_temp :                         // Place last seen
-                           Htw.inverse().linear().leftCols<1>() + Htw.inverse().translation();  // In front of the robot
-                position = (Htw * rBWw).head<2>();
-
-                // Hack Planner:
-                float headingChange = 0;
-                float sideStep      = 0;
-                float speedFactor   = 1;
-                if (useLocalisation) {
-
-                    // Transform kick target to torso space
-                    auto fieldPosition = Eigen::Affine2d(field.position);
-                    Eigen::Affine3d Hfw;
-                    Hfw.translation() =
-                        Eigen::Vector3d(fieldPosition.translation().x(), fieldPosition.translation().y(), 0);
-                    Hfw.linear() = Eigen::AngleAxisd(Eigen::Rotation2Dd(fieldPosition.rotation()).angle(),
-                                                     Eigen::Vector3d::UnitZ())
-                                       .toRotationMatrix();
-
-                    Eigen::Affine3d Htf        = Htw * Hfw.inverse();
-                    Eigen::Vector3d kickTarget = Htf * Eigen::Vector3d(kickPlan.target.x(), kickPlan.target.y(), 0);
-
-                    // //approach point:
-                    Eigen::Vector2d ballToTarget = (kickTarget.head<2>() - position).normalized();
-                    Eigen::Vector2d kick_point   = position - ballToTarget * ball_approach_dist;
-
-                    if (position.norm() > slowdown_distance) {
-                        position = kick_point;
-                    }
-                    else {
-                        speedFactor   = slow_approach_factor;
-                        headingChange = std::atan2(ballToTarget.y(), ballToTarget.x());
-                        sideStep      = 1;
-                    }
+                    case message::behaviour::MotionCommand::Type::WALK_TO_STATE:
+                        determineSimpleWalkPath(ball, field, sensors, kickPlan, fieldDescription);
```

Comment:
imo they are 100% different, but WALK_TO_STATE is implemented wrong, the goal state is never used. WALK_TO_STATE should be renamed in the future to better represent what is does

Reply:
That sounds like something which we would do well to comment (you could even make an issue). Something like `TODO(BehaviourTeam): Reimplement WALK_TO_STATE to <do a thing>, because <reason>, then change this switch`.

Also, don't forget the `[[fallthrough]]` [attribute](https://en.cppreference.com/w/cpp/language/attributes/fallthrough) when you have switch cases which deliberately fall through, so the compiler knows not to warn you because you've done it deliberately.

Reply:
I've implemented the fallthrough with the attribute included as suggested. Just need to double check that this builds and is correct before merging